codalab/lib/bundle_cli.py

```diff
@@ -1621,10 +1622,32 @@ def do_run_command(self, args):
         client, worksheet_uuid = self.parse_client_worksheet_uuid(args.worksheet_spec)
         args.target_spec, args.command = desugar_command(args.target_spec, args.command)
         metadata = self.get_missing_metadata(RunBundle, args)
-
         targets = self.resolve_key_targets(client, worksheet_uuid, args.target_spec)
-        params = {'worksheet': worksheet_uuid}
 
+        if args.interactive:
+            # Disable cl run --interactive on headless systems
+            self._fail_if_headless(args)
+
+            # Fetch bundle locations from the server
+            bundle_uuids = [bundle_target.bundle_uuid for _, bundle_target in targets]
+            bundles_locations = client.get_bundles_locations(bundle_uuids)
+
+            docker_image = metadata.get('request_docker_image', None)
+            if not docker_image:
+                # If a Docker image is not specified, use the default CPU worker image for the interactive session
+                docker_image = self.manager.config['workers']['default_cpu_image']
+
+            # Start an interactive session to allow users to figure out the command to run
+            session = InteractiveSession(docker_image, self.manager, targets, bundles_locations)
```

Comment:
I actually couldn't find a way around this. Since 0 or more targets can be passed in and we have `target_spec` as the positional argument before `command`, not passing in a command will wrongly treat the last target as the command.